include/qengine.hpp

```diff
@@ -25,6 +25,7 @@ typedef std::shared_ptr<QEngine> QEnginePtr;
  */
 class QEngine : public QInterface {
 protected:
+    complex* stateVec;
```

Comment:
Hm I thought this was something that only the CPU variant had...?

Reply:
Actually, no, QEngineOCL also uses it, but it might "live in OCL land," when `LockSync()` hasn't been called. `LockSync()` "brings it back" for general heap access, though, and then `UnlockSync()` preps it for OCL use, again. (This is expected for heap objects, in OpenCL.)